services/agents/registry.go

```diff
@@ -699,7 +698,48 @@ func (r *Registry) runStateChangeHandler(ctx context.Context, agent *pmmAgentInf
 	}
 }
 
-// RequestStateUpdate requests state update on pmm-agent with given ID.
+// SetAllAgentsStatusUnknown goes through all pmm-agents and sets status to UNKNOWN.
+func (r *Registry) SetAllAgentsStatusUnknown(ctx context.Context) error {
+	agents, err := models.FindAgents(r.db.Querier, models.AgentFilters{})
+	if err != nil {
+		return errors.Wrap(err, "failed to get all agents")
+
+	}
+	for _, agent := range agents {
+		// Since we don't set status for external exporters we can skip them.
+		if agent.AgentType == models.ExternalExporterType {
+			continue
+		}
+		// The agents without PMMAgentID set are PMM agents itself.
+		if agent.PMMAgentID == nil && !r.IsConnected(agent.AgentID) {
+			err = r.updateAgentStatusForChildren(ctx, agent.AgentID, inventorypb.AgentStatus_UNKNOWN, 0)
+			if err != nil {
+				return err
+			}
+		}
+	}
+	return nil
+}
+
+func (r *Registry) updateAgentStatusForChildren(ctx context.Context, agentID string, status inventorypb.AgentStatus, listenPort uint32) error {
+	agents, err := models.FindAgents(r.db.Querier, models.AgentFilters{
+		PMMAgentID: agentID,
+	})
+	if err != nil {
+		return errors.Wrap(err, "failed to get pmm-agent's subordinate agents")
```

Comment:
nit: lets keep the terminology consistent
`failed to get pmm-agent's subordinate agents` -> `failed to get pmm-agent's child agents`